Client/src/Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -223,105 +224,39 @@ public Output GetTaskOutputInfo(string taskId)
   {
     using var _ = Logger?.LogFunction();
 
+    var requests = payloadsWithDependencies.Select(pwd =>
+                                                   {
+                                                     var taskRequest = new TaskRequest
+                                                                       {
+                                                                         Payload = UnsafeByteOperations.UnsafeWrap(pwd.Item2),
+                                                                       };
+                                                     taskRequest.DataDependencies.AddRange(pwd.Item3 ?? Enumerable.Empty<string>());
+                                                     taskRequest.ExpectedOutputKeys.Add(pwd.Item1);
+                                                     return taskRequest;
+                                                   })
+                                           .ToList();
```

Comment:
We wanted to avoid this `ToList()` with this implementation

Reply:
With the retry, either you materialize the `requests` list, or you iterate the `payloadsWithDependencies` multiple times.

Reply:
I don't mind either way, I can put it directly but we risk multiple enumeration. Your call